http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/app/CromwellRunnerAppInstall.scala

```diff
@@ -79,6 +84,17 @@ class CromwellRunnerAppInstall[F[_]](config: CromwellRunnerAppConfig,
         AppCreationException(s"Pet not found for user ${params.app.auditInfo.creator}", Some(ctx.traceId))
       )
 
+      maybeProfile <- Try(UUID.fromString(params.billingProfileId.value)) match {
+        case Success(uuid) => bpmClient.getProfile(userToken, uuid)
```

Comment:
Just calling out that this is using the pet token, which is probably not correct (the user might not have access to the billing project/profile). @dvoet has been working on resource-type-admin authorization for service-to-service calls. Should Leo use that instead?

Reply:
I was worried about that and tried it out with a user that didn't have access to the billing project and it worked - but I'm happy to switch if I knew how

Reply:
That’s odd, I wouldn’t expect it to work. :) How did you test exactly — I’m picturing something like this:
1. As user A, create billing project and workspace
2. As user A, make user B a writer on the workspace 
3. As user B, run a workflow in the workspace 

To use the resource-type-admin feature, I think you’d just use the Leo SA token instead of the pet (I think there is a method getLeoToken). There may be some more wiring needed to grant the Leo SA the right permissions on billing profiles.

Reply:
That's exactly what I did and it did for some reason, somehow, work.  I'll look into granting Leo the permissions.

Reply:
Turns out it was because BPM grants read permission to anyone with any access whatsoever to the billing profile.  https://broadworkbench.atlassian.net/browse/WOR-1769 fixed that but left read profile access to avoid changing current behavior.